### DISCUSSION TITLE
Clarification on branches for a beginner (that would be me)

### DISCUSSION THREAD

**Discussion by fabrizioleosanca** [community/community#83137] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Good morning
I just started using GIT and GitHub in Visual Studio 2022 and
Please, I would like someone to give me some clarification on the branches.
I tried to use the branches but I made a mess, I read and reread the Microsoft help but I still didn't understand well. I have an application in a repository, and with the Master branch, I can start a branch from the Master, then if I want to make another branch where should I always start it from Master or from the rom created later? And then in the end I have to bring all the branches together at the Master?
Thank you all
Fabrizio

**Comment by HonkingGoose** [comment, marked as answer]
Read the ProGit 2 book, chapters 1, 2 and 3 to learn the basics of Git. Here's the link, you can read it there for free:

https://git-scm.com/book/en/v2

**Comment by joohndavid** [comment]
[Good morning](https://apklion.com/good-morning-images/) Fabrizio! When working with branches in Git and GitHub, you typically start from the main branch (like "master") and create new branches for specific features or bug fixes. You don't have to always start from the main branch to create new branches; you can create them from the current branch you're on. Once you're done with a branch, you can merge it back into the main branch. Just remember to review your changes and ensure stability before merging. Happy coding!
